scripts/harbor-init/harbor-init.go

```diff
@@ -0,0 +1,236 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+)
+
+const (
+	// Harbor API version
+	HARBOR_API_ENDPOINT = "/api/v2.0"
+)
+
+type Login struct {
+	Url      string
+	Username string
+	Password string
+}
+
+type Projects []struct {
+	Name string `json:"name"`
+}
+
+type ProjectRobots []struct {
+	Name        string `json:"name"`
+	Id          int    `json:"id"`
+	Permissions []struct {
+		Namespace string `json:"namespace"`
+	} `json:"permissions"`
+}
+
+type ResponseRobot struct {
+	Id int `json:"id"`
+}
+
+func request(method string, login Login, endpoint string, body []byte) (*http.Response, []byte, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest(method, login.Url+endpoint, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, nil, err
+	}
+	req.SetBasicAuth(login.Username, login.Password)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, nil, err
+	}
+	return resp, respBody, nil
+}
+
+func main() {
+	urlFlag := flag.String("url", "https://harbor.example.com", "Harbor URL")
+	adminFlag := flag.String("admin", "admin", "Harbor admin username")
+	passwordFlag := flag.String("password", "", "Harbor admin password")
+	projectNameFlag := flag.String("project", "renku-build", "Project name")
+	robotAccountNameFlag := flag.String("robot", "renku-registry-robot", "Robot account name")
+	robotSecretFlag := flag.String("secret", "", "Robot account secret")
+	flag.Parse()
+	// Initialize logger to print to stdout
+	logger := log.New(os.Stdout, "INFO: ", log.Ldate|log.Ltime)
+	login := Login{}
+	// Get login from environment variable
+	if *urlFlag == "" {
+		logger.Fatal("url flag is not set")
+	}
+	login.Url = *urlFlag + HARBOR_API_ENDPOINT
```

Comment:
suggestion(non-blocking): Not sure what happens if `urlFlag` has a slash at the end. Beacuse `HARBOR_API_ENDPOINT` also has a slash at the beginning. Then you would have something like `https://harbor.io//api/v2.0` and hopefully this does not cause trouble. A call to `strings.TrimRight` will address this problem. But feel free to ignore if you wish.

Reply:
added trimright